app/models/job_template.rb

```diff
@@ -93,20 +93,36 @@ def import_parsed(name, text, metadata, options = {})
         existing = self.find_by(:name => name) unless options.delete(:build_new)
         # Don't update if the template already exists, unless we're told to
         return if !options.delete(:update) && existing
-
         template = existing || self.new
         template.sync_inputs(metadata.delete('template_inputs'))
         template.sync_foreign_input_sets(metadata.delete('foreign_input_sets'))
         template.sync_feature(metadata.delete('feature'))
         template.locked = false if options.delete(:force)
-        template.assign_attributes(metadata.merge(:template => text.gsub(/<%\#.+?.-?%>\n?/m, '').strip).merge(options))
-        template.assign_taxonomies if template.new_record?
+        template.assign_attributes(metadata.merge(:template => text.gsub(/<%\#.+?.-?%>\n?/m, '').strip).merge(options).except('organizations', 'locations'))
+        template.assign_taxonomies(metadata) if template.new_record?
         template
       end
       # rubocop:enable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
     end
   end
 
+  def map_metadata(metadata, taxonomy)
+    if metadata[taxonomy]
+      case param
+      when 'locations'
+        metadata[taxonomy].map do |loc|
+          Location.all.map { |db| db.name =~ /^#{loc}/ ? db : nil }
```

Comment:
I assume here we want to find a single (not more) `Location` for each location name  (`metadata['location']`). This makes me think `metadata[taxonomy].map { |loc_name| Location.find(:name => loc_name) }.compact` would be more appropriate. Also why are you matching using a prefix?